### PR TITLE
Implement configurable endpoint setting

### PR DESCRIPTION
**Pull request by tfenster**
This introduces a new setting which allows to set the docker endpoint like the env variable DOCKER_HOST

### CODE REVIEW

**Comment by tfenster** [comment]
Not sure what the problem in the last Travis build is?

File: explorer/dockerExplorer.ts [hunk 1]
```diff
@@ -17,20 +18,24 @@ export class DockerExplorerProvider implements vscode.TreeDataProvider<NodeBase>
...
 
     refresh(): void {
+        docker.refreshEndpoint();
         this._onDidChangeTreeData.fire(this._imagesNode);
         this._onDidChangeTreeData.fire(this._containersNode);
         this._onDidChangeTreeData.fire(this._registriesNode);
     }
 
     refreshImages(): void {
+        docker.refreshEndpoint();
```
Comment (StephenWeatherford) [inline explorer/dockerExplorer.ts]: Better would be to change Configuration.initialize in dockerExtension.ts to approximately this:

    export function initialize() {
        configurationListener = vscode.workspace.onDidChangeConfiguration((e: vscode.ConfigurationChangeEvent) => {
            // notify the language server that settings have change
            client.sendNotification(DidChangeConfigurationNotification.type, { settings: null });

            // Update endpoint if needed
            if (e.affectsConfiguration('docker')) {
                docker.refresh();
            }
        });
    }

**Comment by tfenster** [comment]
@StephenWeatherford I hope I have fixed everything as requested

File: commands/utils/docker-endpoint.ts [hunk 2]
```diff
@@ -9,8 +10,35 @@ class DockerClient {
...
+                newHost = value.slice(0, sep);
+                newPort = Number(value.slice(sep + 1));
+                if (isNaN(newPort)) {
+                    vscode.window.showErrorMessage(errorMessage);
+                } else {
+                    this.endPoint = new Docker({ host: newHost, port: newPort});
+                }
+            }
+        }
+        if (!this.endPoint) {
```
Comment (StephenWeatherford) [inline commands/utils/docker-endpoint.ts]: We missed a case - when changing from a non-empty endpoint to empty endpoint we need to create a new Endpoint:

Please change 
   if (!this.endPoint) { 
to
   if (!this.endPoint || !value) {

File: dockerExtension.ts [hunk 3]
```diff
@@ -172,9 +173,14 @@ namespace Configuration {
...
     export function initialize() {
-        configurationListener = vscode.workspace.onDidChangeConfiguration(() => {
+        configurationListener = vscode.workspace.onDidChangeConfiguration((e: vscode.ConfigurationChangeEvent) => {
             // notify the language server that settings have change
             client.sendNotification(DidChangeConfigurationNotification.type, { settings: null });
+
+            // Update endpoint if needed
+            if (e.affectsConfiguration('docker')) {
+                docker.refreshEndpoint();
+            }
```
Comment (StephenWeatherford) [inline dockerExtension.ts]: Please add following line:

            if (e.affectsConfiguration('docker')) {
                docker.refreshEndpoint();
                vscode.commands.executeCommand("vscode-docker.explorer.refresh");  <<<<<<===========
            }

**Comment by tfenster** [comment]
fixed those as well. Thanks!

**Comment by StephenWeatherford** [comment]
Will be included in the next release.

**Pull request merged**